library/core/src/pin/unsafe_pinned.rs

```diff
@@ -0,0 +1,197 @@
+use crate::marker::{PointerLike, Unpin};
+use crate::ops::{CoerceUnsized, DispatchFromDyn};
+use crate::pin::Pin;
+use crate::{fmt, ptr};
+
+/// The type `UnsafePinned<T>` lets unsafe code violate the rule that `&mut UnsafePinned<T>` may
+/// never alias anything else.
```

Comment:
This is a bit circular; perhaps "`UnsafePinned<T>` provides a way to out of typical aliasing rules; specifically, `&mut UnsafePinned<T>` is not guaranteed to be a unique pointer".